test/integration/instances_test.go

```diff
@@ -589,6 +589,211 @@ func TestInstance_withPG(t *testing.T) {
 	require.Equal(t, inst.PlacementGroup.PlacementGroupPolicy, pg.PlacementGroupPolicy)
 }
 
+func TestInstance_CreateWithReservedIPAddress(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithReservedIPAddress")
+	defer teardown()
+
+	// Reserve an IP for testing
+	reservedIP, err := client.ReserveIPAddress(context.Background(), linodego.ReserveIPOptions{Region: "us-east"})
+	if err != nil {
+		t.Fatalf("Failed to reserve IP: %v", err)
+	}
+	defer func() {
+		err := client.DeleteReservedIPAddress(context.Background(), reservedIP.Address)
+		if err != nil {
+			t.Errorf("Failed to delete reserved IP: %v", err)
+		}
+	}()
+
+	_, instanceTeardown, err := createInstanceWithReservedIP(t, client, reservedIP.Address)
+	if err != nil {
+		t.Fatalf("Error creating instance with reserved IP: %s", err)
+	}
+	defer instanceTeardown()
+
+}
+
+func createInstanceWithReservedIP(
+	t *testing.T,
+	client *linodego.Client,
+	reservedIP string,
+	modifiers ...instanceModifier,
+) (*linodego.Instance, func(), error) {
+	t.Helper()
+
+	createOpts := linodego.InstanceCreateOptions{
+		Label:    "go-test-ins-reserved-ip-" + randLabel(),
+		Region:   "us-east",
+		Type:     "g6-nanode-1",
+		Booted:   linodego.Pointer(false),
+		Image:    "linode/alpine3.17",
+		RootPass: randPassword(),
+		Interfaces: []linodego.InstanceConfigInterfaceCreateOptions{
+			{
+				Purpose:     linodego.InterfacePurposePublic,
+				Label:       "",
+				IPAMAddress: "",
+			},
+		},
+		Ipv4: []string{reservedIP},
+	}
+
+	for _, modifier := range modifiers {
+		modifier(client, &createOpts)
+	}
+
+	instance, err := client.CreateInstance(context.Background(), createOpts)
+	if err != nil {
+		return nil, func() {}, err
+	}
+
+	teardown := func() {
+		if terr := client.DeleteInstance(context.Background(), instance.ID); terr != nil {
+			t.Errorf("Error deleting test Instance: %s", terr)
+		}
+	}
+
+	return instance, teardown, nil
+}
+
+func TestInstance_CreateWithOwnedNonAssignedReservedIP(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithOwnedNonAssignedReservedIP")
+	defer teardown()
+
+	reservedIP, err := client.ReserveIPAddress(context.Background(), linodego.ReserveIPOptions{Region: "us-east"})
+	if err != nil {
+		t.Fatalf("Failed to reserve IP: %v", err)
+	}
+	defer func() {
+		err := client.DeleteReservedIPAddress(context.Background(), reservedIP.Address)
+		if err != nil {
+			t.Errorf("Failed to delete reserved IP: %v", err)
+		}
+	}()
+
+	_, instanceTeardown, err := createInstanceWithReservedIP(t, client, reservedIP.Address)
+	if err != nil {
+		t.Errorf("Unexpected error with owned non-assigned reserved IP: %v", err)
+	} else {
+		instanceTeardown()
```

Comment:
Same comment as above, you should run/defer the teardown before you do any of the checks since it's good practice. (`createInstanceWithReservedIP ` returns an empty function for `instanceTeardown ` if it encounters an error so it should be safe to call `defer` on it)

Reply:
Added defer statement immediately after the creation statement.